reconcile/saas_auto_promotions_manager/integration.py

```diff
@@ -133,11 +139,16 @@ def init_external_dependencies(
     deployment_state = PromotionState(
         state=init_state(integration=OPENSHIFT_SAAS_DEPLOY, secret_reader=secret_reader)
     )
+    s3_exporter = S3Exporter(
+        state=init_state(integration=QONTRACT_INTEGRATION, secret_reader=secret_reader),
```

Comment:
`state` requires call `state.cleanup()` after use to close aws client.

Reply:
Good catch! Fixing here https://github.com/app-sre/qontract-reconcile/pull/4162